src/platform/tigerlake/include/platform/lib/shim.h

```diff
@@ -139,6 +139,15 @@
 #define GPDMA_CLKCTL(x)		(0x78404 + x*0x100)
 #define GPDMA_FDCGB		(0x1 << 0)
 
+/* I2S SHIM Registers */
+#define I2SLCTL			0x71C04
+
+/* SPA register should be set for each I2S port and DSP should
+ * wait for CPA to be set
```

Comment:
which we are not doing this patch?

Reply:
Thanks, for spotting, fixed. I guess spin-check without timeout is ok for such things, because it would mean that HW is broken if it fails to set CPA.

Reply:
@plbossart Actually have to postpone doing it with CPA checks as it does not work as described. Need more info about HW, cos correct flow is not as it's written.

Reply:
CPA is not set even within few seconds after setting SPA, however it's set meantime after setting sometihing in ssp params, maybe there are some unknown dependencies on clock gating etc